sklearn/feature_extraction/text.py

```diff
@@ -1725,7 +1727,12 @@ def transform(self, X, copy=True):
         return X
 
     def _more_tags(self):
-        return {"X_types": ["2darray", "sparse"]}
+        return {
+            "X_types": ["2darray", "sparse"],
+            # FIXME: np.float16 could be preserved if _inplace_csr_row_normalize_l2
+            # accepted it.
+            "preserves_dtype": [np.float64, np.float32],
```

Comment:
> Should we have some tests to check the preservation?

@OmarManzoor When "preserves_dtype" is added to a estimator tag, it will run the following test in the common tests automatically:

https://github.com/scikit-learn/scikit-learn/blob/d418e79a41d5d4de167114d88465a0b47b0d6a2c/sklearn/utils/estimator_checks.py#L1927

Reply:
Oh I see. Thanks for the clarification!